binderhub/registry.py

```diff
@@ -192,7 +192,7 @@ def get_image_manifest(self, image, tag):
         # first, get a token to perform the manifest request
         if self.token_url:
             auth_req = httpclient.HTTPRequest(
-                url_concat(self.token_url, {"scope": "repository:{}:pull".format(image)}),
+                url_concat(self.token_url, {"scope": "repository:{}:pull".format(image), "service": "container_registry"}),
```

Comment:
For one of the clusters on mybinder.org we [use a Harbor registry](
https://github.com/jupyterhub/mybinder.org-deploy/blob/d91b634ccb8f1fe5fa2bb02551ad576f7f7e516d/config/ovh.yaml#L12) and have a URL like: https://dlqpwel7.gra5.container-registry.ovh.net/service/token?service=harbor-registry as the `token_url`. What happens if the token URL already specifies a service parameter and we try and set it again from here?

We should try and not break existing users, in which case maybe we should add a section [below here](https://github.com/jupyterhub/binderhub/blob/master/doc/setup-binderhub.rst#if-you-are-using-azure-container-registry) specifically for using a GitLab registry.

Reply:
You're right of course. Not sure why i didn't just think of passing it as a query arg. I'll drop that commit/